File: Python/PythagoreanTriple/pythagorean_triple.py
# This program determines if a set of three numbers is a Pythagorean triple

# Define variables
side_one = 0
side_two = 0
hypotenuse = 0

# For each value of hypotenuse from 1 to 20
for hypotenuse in range( 1, 21 ):
    # For each value of side_two from 1 to 20
    for side_two in range( 1, 21 ):
        # For each value of side_one from 1 to 20
        for side_one in range( 1, 21 ):
            # If side_one squared plus side_two squared equals hypotenuse squared
            if side_one ** 2 + side_two ** 2 == hypotenuse ** 2:
                # Print a message
                print( "{}, {}, and {} are a Pythagorean triple.".format(
                        side_one, side_two, hypotenuse ) )

File: C/PythagoreanTriple/pythagoreanTriple.c
/* This program determines if a set of three numbers is a pythagorean triple */
#include <stdio.h>

/* Function main begins program execution */
int main( void )
{
    /* For every value of hypotenuse */
    for ( int hypotenuse = 1; hypotenuse <= 500; hypotenuse++ )
    {
        /* For every value of sideOne */
        for ( int sideOne = 1; sideOne <= 500; sideOne++ )
        {
            /* For every value of sideTwo */
            for ( int sideTwo = 1; sideTwo <= 500; sideTwo++ )
            {
                /* If sideOne squared plus sideTwo squared equals hypotenuse squared */
                if ( ( sideOne * sideOne ) + ( sideTwo * sideTwo ) == ( hypotenuse * hypotenuse ) )
                {
                    /* Print the result */
                    printf( "%d, %d, and %d are a Pythagorean triple.\n", sideOne, sideTwo, hypotenuse );
                } /* end if */
            } /* end for */
        } /* end for */
    } /* end for */

    return 0; /* Indicate the program ended successfully */
} /* end function main */
